### PR TITLE
Fine grained configuration possibility for Metadata Exporter

### PR DESCRIPTION
**Pull request by attila-kiss-it**
Fixes for #797

### CODE REVIEW

File: querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/NamingStrategy.java [hunk 1]
```diff
@@ -143,4 +157,10 @@
      */
     String normalizeSchemaName(String schemaName);
 
+    boolean shouldGenerateClass(SchemaAndTable schemaAndTable);
```
Comment (timowest) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/NamingStrategy.java]: javadocs needed here

File: querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/AbstractNamingStrategy.java [hunk 2]
```diff
@@ -122,4 +124,25 @@ public void setReservedSuffix(String reservedSuffix) {
         this.reservedSuffix = reservedSuffix;
     }
 
+    @Override
+    public String getPackage(String basePackage, SchemaAndTable schemaAndTable) {
+        return basePackage;
```
Comment (timowest) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/AbstractNamingStrategy.java]: shouldn't this delegate to appendSchema?

File: querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java [hunk 3]
```diff
@@ -372,15 +387,19 @@ private void serialize(EntityType type) {
         try {
             String fileSuffix = createScalaSources ? ".scala" : ".java";
 
+            String schemaName = (String) type.getData().get("schema");
+            String tableName = (String) type.getData().get("table");
+            SchemaAndTable schemaAndTable = new SchemaAndTable(schemaName, tableName);
```
Comment (timowest) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java]: the SchemaAndTable instance which is created in the previous could be given as an argument, then the normalized schema and table name would be used in both cases.

**Comment by timowest** [comment]
This looks really good now, I found one thing could be still improved.

File: querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java [hunk 4]
```diff
@@ -123,9 +123,10 @@ public MetaDataExporter() { }
     protected EntityType createEntityType(@Nullable String schemaName, String tableName,
```
Comment (timowest) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java]: provide SchemaAndTable as argument

**Comment by timowest** [comment]
Looks good now. @attila-kiss-it Thanks a lot for the contribution!

**Comment by Shredder121** [comment]
Sorry for the delay, thank you @attila-kiss-it for the contribution.

**Pull request merged**